### PR TITLE
chore(Transactions): AND-1372 Create single datamanager for handling all cryptocurrency transactions

### PR DESCRIPTION
**Pull request by ditn**

### CODE REVIEW

File: balance/src/main/kotlin/info/blockchain/balance/CryptoValue.kt [hunk 1]
```diff
@@ -38,8 +38,15 @@ data class CryptoValue(
         }
 
         fun bitcoinFromSatoshis(satoshi: Long) = CryptoValue(CryptoCurrency.BTC, satoshi.toBigInteger())
+        fun bitcoinFromSatoshis(satoshi: BigInteger) = CryptoValue(CryptoCurrency.BTC, satoshi)
         fun bitcoinCashFromSatoshis(satoshi: Long) = CryptoValue(CryptoCurrency.BCH, satoshi.toBigInteger())
+        fun bitcoinCashFromSatoshis(satoshi: BigInteger) = CryptoValue(CryptoCurrency.BCH, satoshi)
+        @Deprecated(
```
Comment (westonal) [inline balance/src/main/kotlin/info/blockchain/balance/CryptoValue.kt]: Fair enough

File: testutils/src/main/kotlin/com/blockchain/testutils/Currencies.kt [hunk 2]
```diff
@@ -18,6 +18,6 @@ private fun Number.numberToBigDecimal(): BigDecimal =
         else -> throw NotImplementedError(this.javaClass.name)
     }
 
-fun Double.bitcoin() = CryptoValue.bitcoinFromMajor(toBigDecimal())
-fun Double.ether() = CryptoValue.etherFromMajor(toBigDecimal())
-fun Double.bitcoinCash() = CryptoValue.bitcoinCashFromMajor(toBigDecimal())
+fun Number.bitcoin() = CryptoValue.bitcoinFromMajor(numberToBigDecimal())
+fun Number.ether() = CryptoValue.etherFromMajor(numberToBigDecimal())
+fun Number.bitcoinCash() = CryptoValue.bitcoinCashFromMajor(numberToBigDecimal())
```
Comment (westonal) [inline testutils/src/main/kotlin/com/blockchain/testutils/Currencies.kt]: 👍

File: core/src/test/java/com/blockchain/datamanagers/TransactionSendDataManagerTest.kt [hunk 3]
```diff
@@ -0,0 +1,420 @@
...
+        // Act
+        subject.executeTransaction(amount, destination, account, feeOptions, FeeType.Priority)
+            .test()
+        // Assert
+        verify(sendDataManager).getUnspentOutputs("XPUB")
+        verify(sendDataManager).getSpendableCoins(
+            unspentOutputs,
+            BigInteger.TEN,
+            feeOptions.priorityFee.toBigInteger()
+        )
```
Comment (westonal) [inline core/src/test/java/com/blockchain/datamanagers/TransactionSendDataManagerTest.kt]: Same comment on these still. They don't test anything we care about, i.e. behaviour, just implementation. The "entire flow" tests cover this.

Comment (ditn) [reply core/src/test/java/com/blockchain/datamanagers/TransactionSendDataManagerTest.kt]: Oh I thought I'd done this. Let me triple check my cherry-pick.

**Pull request merged**